simpl/logging.py

```diff
@@ -0,0 +1,180 @@
+# Copyright (c) 2011-2015 Rackspace US, Inc.
+# All Rights Reserved.
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+"""Logging Boilerplate.
+
+Implements:
+- integration with :mod:`config` for configuring logging.
+- enhanced handling of extra data and output formatting.
+
+
+Usage with config:
+
+- initialization
+
+    from simpl import config
+    from simpl import logging
+
+    # Add logging.OPTIONS to your options when implementing config
+
+
+Usage in module:
+
+    from simpl import logging  # instead of `import logging`
+
+    LOG = logging.getLogger(__name__)
+
+
+"""
+from __future__ import absolute_import
+
+import logging
+import os
+import sys
+
+from simpl import config
+
+OPTIONS = [
+    #
+    # Verbosity, debugging, and monitoring
+    #
+    config.Option("--logconfig",
+                  help="Optional logging configuration file"),
+    config.Option("-d", "--debug",
+                  default=False,
+                  action="store_true",
+                  help="turn on additional debugging inspection and "
+                  "output including full HTTP requests and responses. "
+                  "Log output includes source file path and line "
+                  "numbers"),
+    config.Option("-v", "--verbose",
+                  default=False,
+                  action="store_true",
+                  help="turn up logging to DEBUG (default is INFO)"),
+    config.Option("-q", "--quiet",
+                  default=False,
+                  action="store_true",
+                  help="turn down logging to WARN (default is INFO)"),
+]
+
+
+def getLogger(name):  # pylint: disable=C0103
```

Comment:
What about just pointing this function to `logging.getLogger` that way you get the `__doc__` and everything else?

```python
getLogger = logging.getLogger
```